platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrClientFactoryImpl.java

```diff
@@ -32,7 +41,11 @@
             factory = new HttpSolrClientFactory();
         }
 
-        return factory.newClient(core);
+        return newClient(factory, core);
     }
 
+    // Package-private for unit testing.
```

Comment:
Was this not possible with constructor injection? Maybe we could inject a `Supplier` like last time?